### ISSUE TITLE
Bug: Parser should error when both `projectService` and `project` are set

### ISSUE THREAD

**Issue by MariaSolOs** [typescript-eslint/typescript-eslint#11319]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.
- [x] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [x] I have [read the FAQ](https://typescript-eslint.io/linting/troubleshooting) and my problem is not listed.

### Relevant Package

parser

### Repro Code

```TypeScript
N/A. This issue is about a confusing configuration behavior.
```

### ESLint Config

```javascript
// base-config.js
module.exports = {
  plugins: [ "@typescript-eslint" ],
  parser: "@typescript-eslint/parser",
  parserOptions: { projectService: true }
}

// .eslintrc.js
module.exports = {
  extends: "./base-config.js",
  parserOptions: { project: "./tsconfig.eslint.json" }
}
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

The docs aren't clear of what the parser will use when both `projectService` and `project` are set. This can be particularly confusing when extending a configuration that sets `parserOptions` which are later combined into a single merged object. I would expect for the parser to throw an explicit error in such misconfiguration scenarios, or have a well-documented explanation of what happens when both settings are provided.

### Actual Result

`eslint --print-config` shows that the merged configuration has `{ project: "./tsconfig.eslint.json", projectService: true }`, and it's not clear how the TypeScript program will be discovered.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `^8.31.0` |
| `@typescript-eslint/parser`            | `^8.31.0` |
| `TypeScript`                           | `5.8.3` |
| `ESLint`                               | `^8.57.0` |
| `node`                                 | `22.14.0` |

**Comment by JoshuaKGoldberg** [comment]
This is a good point. For most misconfiguration items I would normally point to #6403 and say we should address that holistically. But this _"both `project` and `projectService`"_ case has come up a few times. IMO it'd be better to get a quicker, tailored fix now and not wait for #6403's breaking change some major version in the future.

cc @typescript-eslint/triage-team - WDYT?

**Comment by JoshuaKGoldberg** [comment]
Coming over from #11333: there are some users for whom this would be a breaking change. Maybe we should temporarily give users a way to disable the new error? `TYPESCRIPT_ESLINT_IGNORE_PROJECT_AND_PROJECT_SERVICE_ERROR` or some similarly unwieldy name?

My leaning is to add that in now and remove it as a breaking change in our next major version.

**Comment by MariaSolOs** [comment]
> Coming over from [#11333](https://github.com/typescript-eslint/typescript-eslint/pull/11333): there are some users for whom this would be a breaking change. Maybe we should temporarily give users a way to disable the new error? `TYPESCRIPT_ESLINT_IGNORE_PROJECT_AND_PROJECT_SERVICE_ERROR` or some similarly unwieldy name?
> 
> My leaning is to add that in now and remove it as a breaking change in our next major version.

Yeah that's a good idea. I'm happy to add this to my PR if the team is onboard with incorporating this behavior :)

**Comment by JoshuaKGoldberg** [comment]
Following up, I think we're good to accept this as a standalone change. FYI @MariaSolOs, thanks for waiting on us!

**Comment by adrian-heath** [comment]
This has broken the use of this plugin within my angular 20 build.  I am not sure exactly why since the eslint config only specifies projectService but my angular solution does include both standalone library projects as well as the main solution.  The error does not  output the values of the project and projectService so it is proving a bit hard to track down.

**Comment by JoshuaKGoldberg** [comment]
Huh, GitHub didn't auto-close this issue when #11333 was merged. But this was fixed by #11333.

**Comment by JoshuaKGoldberg** [comment]
@adrian-heath could you please post a new issue with a standalone reproduction? We can't help you from just a high-level description of your issue - we'd need to see specifics of your build.

You might also want to try setting the `TYPESCRIPT_ESLINT_IGNORE_PROJECT_AND_PROJECT_SERVICE_ERROR` environment variable. It's not documented (yet?) on the website but it should suppress these kinds of failures.
